### ISSUE TITLE
Nonce mismatch between public and private nonces

### ISSUE THREAD

**Issue by roderik** [hyperledger/besu#1469]
### Description

I'm having a strange issue with private transaction nonces. 

```
│ 2020-10-19 09:33:03.052+00:00 | EthScheduler-Workers-2 | ERROR | PrivacyPrecompiledContract | Failed to process private transaction 0x1dbdc23939ede1367558532130d0875a3d47cb1cf7d0adc71c615f902a362ea2: Private Transaction nonce 67, does not match sender account nonce 0.                                                          │
```

This happens in ["my" version](https://github.com/settlemint/besu-privacy-examples/blob/main/scripts/multinode.ts) of https://github.com/ConsenSys/web3js-eea/blob/master/example/multiNodeExample/storeValueFromNode1.js 

I'm not setting the nonce manually, I wait between each call for it to be mined, so the issues about concurrent tx do not apply. 

The result is that in step 7, I get a return value of 0 instead of 5000

![output](https://user-images.githubusercontent.com/16780/96431368-b244fe80-1203-11eb-8f9f-31bab2feb03f.png)

Any idea where to look?

**Bug Triage**

Severity: Critical (breaks a key privacy behaviour)
Frequency: Remote (we haven't seen this happening before in any of other privacy networks)
Assessed Bug Priority: P3

**Comment by lucassaldanha** [comment]
Hi @roderik ! Thanks for raising this issue. Let's see if we can get to the bottom of it.

I'll try to run your script later, while I look at it, would you be able to tell me what is the nonce for user1 account in the privacy group that you are deploying the contract? You can use this method https://besu.hyperledger.org/en/latest/Reference/API-Methods/#priv_geteeatransactioncount

```
curl -X POST --data '{"jsonrpc":"2.0","method":"priv_getEeaTransactionCount","params":["<user1_ethereum_address>", "< node1_orion_key>", ["node2_orion_key"]], "id":1}' http://<node_ip>:<port>
```

**Comment by roderik** [comment]
Ok, some more info that might help

1. I had already destroyed this network, so I could not do the debugging steps
2. I created a new network, and it worked

So thinking about the difference between the two, I had the initial network setup wrong and the Orion nodes were not connected. I messed up some ports and NAT announcements and /knownhosts on Orion was empty. While the Besu nodes were connected to their own Orion and to each other. In that faulty state, I ran the test script several times. 

So my guess, is that the way it keeps track of nonces in that scenario is a little off.
